Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/event/dao/DatabaseEventLoggerDao.java

```diff
@@ -69,220 +74,126 @@ private SingletonHolder() {
         }
     }
 
-    /**
-     * Get an instance of DatabaseEventLoggerDao.
-     *
-     * @return singleton instance of DatabaseEventLoggerDao
-     */
     public static DatabaseEventLoggerDao getInstance() {
         LOG.debug("getInstance()...");
         return SingletonHolder.INSTANCE;
     }
 
-    /**
-     * Insert an event in the database.
-     *
-     * @param databaseEvent to be added to the database
-     * @return true if the event was successfully inserted.
-     */
     public boolean insertEvent(final DatabaseEvent databaseEvent) {
-
-        SessionFactory sessionFactory = null;
-        Session session = null;
-        Transaction tx = null;
-        boolean result = true;
-
-        try {
-            sessionFactory = getSessionFactory();
-            if (sessionFactory != null) {
-                session = sessionFactory.openSession();
-                tx = session.beginTransaction();
-
-                session.persist(databaseEvent);
-
-                tx.commit();
-            }
-
-        } catch (final HibernateException e) {
-            result = false;
-            transactionRollback(tx);
-            LOG.error("Exception during insertion caused by : {}", e.getMessage(), e);
-        } finally {
-            closeSession(session);
-        }
-
-        return result;
+        return save(getSession(), databaseEvent);
```

Comment:
Do we need to release session in here?

Reply:
the genericdbutil--save method call saveall and saveall does a final-close-session